### DISCUSSION TITLE
Add link button to dropdown

### DISCUSSION THREAD

**Discussion by cgolankotter** [twilio-labs/paste#3316] [General] [answered]
Hi Team,

We’re looking to add/include a button link in the dropdown on the [Sender ID page](https://www.figma.com/file/MS7xFr7zi11VX4RnvoB4b3/2023-Q2---Sender-ID----Phase-1?type=design&node-id=590-434978&mode=design&t=OuXAiigCMSUoL0pS-4), so users can view all existing options and create new ones as required.

- We’d like to keep this action in context inside the dropdown vs. outside the field, since we're prioritizing the "Add New" over the "Twilio's Generic sender" option. (See mockup)
![Service settings - Sender ID Expanded](https://github.com/twilio-labs/paste/assets/107283881/ce7f446c-c4c8-4957-831c-c86f8801b3ef)

- Messaging team have previously created a similar custome-component in the [Whatsapps Senders section](https://console.twilio.com/us1/develop/sms/senders/whatsapp-senders) (See screenshot)
<img width="2554" alt="Messaging Ref" src="https://github.com/twilio-labs/paste/assets/107283881/ac03fd52-41a0-456b-9b06-768138e34547">

Is there currently something similar to this / guidance on what existing component could fit? Or if not to maybe consider this option in the future?

Appreciate your guidance and help🙏
Thanks!

(cc @fzhao@twilio.com)

**Comment by SiTaggart** [comment, marked as answer]
Hi @cgolankotter 

I suspect the Messaging example is a lot less custom than you think. You can compose any content into a combobox option as long as it's not a focusable action like a button or anchor. 

But if you wanted to add some blue text with an icon, that when selected the combobox redirected you to a new page, that's totally doable with what we currently have today. It's just an engineering task to create the custom content, and respond to the selection event to perform the redirection.

hope that helps.

**Reply by cgolankotter** [reply]
Hi @SiTaggart,
Thanks so much! This is helpful.
I'll share it with our devs.
